src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarEntry.cs

```diff
@@ -138,10 +138,12 @@ public string LinkName
         /// <remarks>The value in this field has no effect on Windows platforms.</remarks>
         public UnixFileMode Mode
         {
-            get => (UnixFileMode)_header._mode;
+            // Some paths do not use the setter, and we want to return valid UnixFileMode.
+            // This mask only keeps the least significant 12 bits.
```

Comment:
I wonder if not using the setter everywhere is correct. it seems somewhat inconsistent and we could've thrown earlier.

Reply:
Good point, I am not sure if this is intentional. I didn't wanted to go for larger refactoring at this point. There is also an existing helper: https://github.com/dotnet/runtime/blob/88aea31185824d30cb1a2edcfe2d0472000d9e22/src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHelpers.Unix.cs#L55, but for some reason we don't use it in the path OP was hitting the exception.